File: Dot examples/ASP Book Examples/Book Examples/Chapter27/CS/App_Code/RSSHandler.cs
using System;
using System.Web;
using System.Net;
using System.IO;

namespace AspNetUnleashed
{
    public class RSSHandler : IHttpAsyncHandler
    {
        private HttpContext _context;
        private WebRequest _request;

        public IAsyncResult BeginProcessRequest(HttpContext context, AsyncCallback cb, object extraData)
        {
            // Store context
            _context = context;

            // Initiate call to RSS feed
            _request = WebRequest.Create("http://msdn.microsoft.com/asp.net/rss.xml");
            return _request.BeginGetResponse(cb, extraData);
        }

        public void EndProcessRequest(IAsyncResult result)
        {
            // Get the RSS feed
            string rss = String.Empty;
            WebResponse response = _request.EndGetResponse(result);
            using (response)
            {
                StreamReader reader = new StreamReader(response.GetResponseStream());
                rss = reader.ReadToEnd();
            }
            _context.Response.Write(rss);
        }

        public bool IsReusable
        {
            get { return true; }
        }

        public void ProcessRequest(HttpContext context)
        {
            throw new Exception("The ProcessRequest method is not implemented.");
        }
    }
}

File: Dot examples/ASP Book Examples/Book Examples/Chapter27/VB/App_Code/RSSHandler.vb
Imports System
Imports System.Web
Imports System.Net
Imports System.IO

Namespace AspNetUnleashed
    Public Class RSSHandler
        Implements IHttpAsyncHandler

        Private _context As HttpContext
        Private _request As WebRequest

        Public Function BeginProcessRequest(ByVal context As HttpContext, ByVal cb As AsyncCallback, ByVal extraData As Object) As IAsyncResult Implements IHttpAsyncHandler.BeginProcessRequest
            ' Store context
            _context = context

            ' Initiate call to RSS feed
            _request = WebRequest.Create("http://msdn.microsoft.com/asp.net/rss.xml")
            Return _request.BeginGetResponse(cb, extraData)
        End Function

        Public Sub EndProcessRequest(ByVal result As IAsyncResult) Implements IHttpAsyncHandler.EndProcessRequest
            ' Get the RSS feed
            Dim rss As String = String.Empty
            Dim response As WebResponse = _request.EndGetResponse(result)
            Using response
                Dim reader As New StreamReader(response.GetResponseStream())
                rss = reader.ReadToEnd()
            End Using
            _context.Response.Write(rss)
        End Sub

        Public ReadOnly Property IsReusable() As Boolean Implements IHttpHandler.IsReusable
            Get
                Return True
            End Get
        End Property

        Public Sub ProcessRequest(ByVal context As HttpContext) Implements IHttpHandler.ProcessRequest
            Throw New Exception("The ProcessRequest method is not implemented.")
        End Sub
    End Class
End Namespace
